psi4/driver/procrouting/scf_proc/scf_iterator.py

```diff
@@ -0,0 +1,766 @@
+            calc_type = core.PCM.CalcType.NucAndEle
+        Dt = self.Da().clone()
+        Dt.add(self.Db())
+        _, Vpcm = self.get_PCM().compute_PCM_terms(Dt, calc_type)
+        self.push_back_external_potential(Vpcm)
+
+    # recompute the Fock matrices as they are modified during the SCF
+    #   iteration and might need to be dumped to checkpoint later
+    self.form_F()
+
+    # Properties
+    #  Comments so that autodoc utility will find these PSI variables
+    #  Process::environment.globals["SCF DIPOLE X"] =
+    #  Process::environment.globals["SCF DIPOLE Y"] =
+    #  Process::environment.globals["SCF DIPOLE Z"] =
+    #  Process::environment.globals["SCF QUADRUPOLE XX"] =
+    #  Process::environment.globals["SCF QUADRUPOLE XY"] =
+    #  Process::environment.globals["SCF QUADRUPOLE XZ"] =
+    #  Process::environment.globals["SCF QUADRUPOLE YY"] =
+    #  Process::environment.globals["SCF QUADRUPOLE YZ"] =
+    #  Process::environment.globals["SCF QUADRUPOLE ZZ"] =
+
+    # Orbitals are always saved, in case an MO guess is requested later
+    # save_orbitals()
+
+    # Shove variables into global space
+    for k, v in self.variables().items():
+        core.set_variable(k, v)
+
+    self.finalize()
+
+    core.print_out("\nComputation Completed\n")
+
+    return energy
+
+
+def scf_print_energies(self):
+    enuc = self.get_energies('Nuclear')
+    e1 = self.get_energies('One-Electron')
+    e2 = self.get_energies('Two-Electron')
+    exc = self.get_energies('XC')
+    ed = self.get_energies('-D')
+    evv10 = self.get_energies('VV10')
+    eefp = self.get_energies('EFP')
+    epcm = self.get_energies('PCM Polarization')
+
+    hf_energy = enuc + e1 + e2
+    dft_energy = hf_energy + exc + ed + evv10
+    total_energy = dft_energy + eefp + epcm
+
+    core.print_out("   => Energetics <=\n\n")
+    core.print_out("    Nuclear Repulsion Energy =        {:24.16f}\n".format(enuc))
+    core.print_out("    One-Electron Energy =             {:24.16f}\n".format(e1))
+    core.print_out("    Two-Electron Energy =             {:24.16f}\n".format(e2))
+    if self.functional().needs_xc():
+        core.print_out("    DFT Exchange-Correlation Energy = {:24.16f}\n".format(exc))
+        core.print_out("    Empirical Dispersion Energy =     {:24.16f}\n".format(ed))
+        core.print_out("    VV10 Nonlocal Energy =            {:24.16f}\n".format(evv10))
+    if core.get_option('SCF', 'PCM'):
+        core.print_out("    PCM Polarization Energy =         {:24.16f}\n".format(epcm))
+    if hasattr(self.molecule(), 'EFP'):
+        core.print_out("    EFP Energy =                      {:24.16f}\n".format(eefp))
+    core.print_out("    Total Energy =                    {:24.16f}\n".format(total_energy))
+
+    self.set_variable('NUCLEAR REPULSION ENERGY', enuc)
+    self.set_variable('ONE-ELECTRON ENERGY', e1)
+    self.set_variable('TWO-ELECTRON ENERGY', e2)
+    if abs(exc) > 1.0e-14:
```

Comment:
Can we change this to `self.functional().needs_xc()`. Will make this logic cleaner.